File: core/src/main/java/com/palomiklo/streampulse/thread/CustomThreadFactory.java
package com.palomiklo.streampulse.thread;

import java.util.concurrent.ThreadFactory;

import jakarta.annotation.Nullable;

public class CustomThreadFactory {
    @Nullable
    public static final ThreadFactory streamPulseThreadFactory = new ThreadFactory() {
        private final String namePattern = "stream-pulse-thread-%d";
        private int count = 0;

        @Override
        public Thread newThread(Runnable r) {
            Thread thread = Thread.ofVirtual().unstarted(r);
            thread.setName(String.format(namePattern, count++));
            return thread;
        }
    };
}

File: core/src/main/kotlin/com/palomiklo/streampulse/thread/CustomThreadFactory.kt
package com.palomiklo.streampulse.thread

import java.util.concurrent.ThreadFactory

object CustomThreadFactory {
    val streamPulseThreadFactory: ThreadFactory = object : ThreadFactory {
        private val PREFIX = "stream-pulse-thread"
        private var count = 0
        override fun newThread(r: Runnable): Thread = Thread.ofVirtual().unstarted(r).apply { setName("$PREFIX-$count++") }
    }
}